storage/src/main/java/tech/pegasys/artemis/storage/HistoricalChainData.java

```diff
@@ -50,6 +60,16 @@ public HistoricalChainData(final EventBus eventBus) {
         slot, new GetLatestFinalizedBlockAtSlotRequest(slot), QUERY_TIMEOUT);
   }
 
+  public SafeFuture<Optional<BeaconState>> getFinalizedStateAtSlot(final UnsignedLong slot) {
+    return stateAtSlotRequests.sendRequest(
+        slot, new GetFinalizedStateAtSlotRequest(slot), QUERY_TIMEOUT);
+  }
+
+  public SafeFuture<Optional<BeaconState>> getFinalizedStateAtBlock(final Bytes32 block) {
```

Comment:
Probably `getFinalizedStateByBlockRoot(final Bytes32 blockRoot)` and that terminology change flows through the various events etc.